docs/features/dynamic_tasks.md

```diff
@@ -0,0 +1,100 @@
+# **Dynamic tasks**
+
+DAG Factory supports Airflow’s
+[Dynamic Task Mapping](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/dynamic-task-mapping.html),
+enabling workflows to dynamically create tasks at runtime. This means the number of tasks does not need to be
+
+## **Example: Defining Dynamic Tasks**
+
+Below is an example configuration for implementing dynamic tasks using DAG Factory:
+
+```title="example_dynamic_task_mapping.yml"
+--8<-- "dev/dags/example_dynamic_task_mapping.yml"
+```
+
+### **Explanation of the Configuration**
+
+1. **`request` Task**:
+    - Generates a list of items using the `make_list` function from the [expand_tasks.py](https://github.com/astronomer/dag-factory/blob/main/dev/dags/expand_tasks.py) module.
+    - This task serves as the input provider for the dynamically mapped tasks.
+
+2. **`process` Task**:
+    - Dynamically generates one task for each item in the list produced by the `request` task.
+    - The expand argument is used to create these tasks at runtime, with `request.output` supplying the input list.
+    - Additionally, the `partial` argument is used to specify fixed parameters (`op_kwargs`) that are applied to all dynamically generated tasks.
+
+### **How It Works**
+
+- **Dynamic Task Creation**:
+    The `expand` keyword allows the process task to spawn multiple tasks at runtime, each processing a single item from
+the list output of the `request` task.
+
+- **Fixed Parameters**:
+    The partial keyword ensures that common parameters, such as `fixed_param`, are passed to every dynamically created
+task instance.
+
+### **Benefits of Dynamic Task Mapping with DAG Factory**
```

Comment:
I'm wondering if we need **?

Reply:
same question for other heading too

Reply:
yeah, maybe we don't need it bold. Removed it from headings. thanks.